### PR TITLE
GET /heights returns block difference b/w 2 endpoints

### PR DESCRIPTION
**Pull request by dimroc**
```
NAME:
   main - CLI for EthBlockComparer: ethblockcomparer <ethereum rpc address 1> <ethereum rpc address 2>

USAGE:
   main [global options] command [command options] [arguments...]

VERSION:
   1.0.1

COMMANDS:
     help, h  Shows a list of commands or help for one command

GLOBAL OPTIONS:
   --threshold value, -t value   Difference in the block height before returning error (default: 20)
   --insecure         Controls whether a client verifies the server's certificate chain and host name
   --help, -h                    show help
   --version, -v                 print the version
```

Example CLI command:

```
go run main.go localhost:18545 https://ropsten.geth.chain.link:8545
```

Example output:

```json
{
  "difference": "1",
  "endpoints": [
    { "number": "0x1", "url": "http://127.0.0.1:59503" },
    { "number": "0x2", "url": "http://127.0.0.1:59504" }
  ],
  "threshold": "2"
}
```

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+func (clientMock) Call(result interface{}, method string, args ...interface{}) error { return nil }
+func (clientMock) Endpoint() string {
+	return "http://clientmock.com"
+}
+
+type clientErrorMock struct{}
+
+func (clientErrorMock) Call(result interface{}, method string, args ...interface{}) error {
+	return errors.New("clientErrorMock")
```
Comment (dimroc) [inline main_test.go]: @j16r would you recommend I switch these hand rolled mocks to https://github.com/golang/mock or leave them as is?

Comment (j16r) [reply main_test.go]: yeah!

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.PHONY: all build
```
Comment (j16r) [inline Makefile]: no `build` target

File: main.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	return endpoint
+}
+
+func (hc *heightsController) Index(c *gin.Context) {
+	var latest1, latest2 block
+	err1 := hc.client1.Call(&latest1, "eth_getBlockByNumber", "latest", false)
+	err2 := hc.client2.Call(&latest2, "eth_getBlockByNumber", "latest", false)
+	merr := multierr.Combine(err1, err2)
+	if merr != nil {
+		log.Println("Error:", merr)
```
Comment (j16r) [inline main.go]: Meant to leave this in?

File: main.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+		c.JSON(502, gin.H{"error": merr.Error()})
+		return
+	}
+
+	difference := big.NewInt(0)
```
Comment (j16r) [inline main.go]: I feel like 113-125 could be tested in isolation as a pure function?

File: main.go (hunk 3 above)
Comment (dimroc) [reply main.go]: I did. By default, `gin` doesn't output logs with
```
c.JSON(502, gin.H{"error": merr.Error()})
```

but now that I think about it, I should probably use `c.AbortWith...` so I'll explore that idea.

**Review by j16r — APPROVED** [review]

**Pull request merged**